aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AmplifyCredential.kt

```diff
@@ -113,11 +113,10 @@ internal data class CognitoUserPoolTokens(
     override fun equals(other: Any?): Boolean {
         return if (super.equals(other)) {
             true
-        } else if (other == null || javaClass != other.javaClass) {
+        } else if (other == null || javaClass != other.javaClass || other !is CognitoUserPoolTokens) {
```

Comment:
Wouldn't javaClass != other.javaClass be enough to rule out other not being of type CognitoUserPoolTokens?

Reply:
It is but it doesn't add the cast safety required for the else block "as".